python/cudf/cudf/core/dataframe.py

```diff
@@ -1488,6 +1488,16 @@ def reset_index(self, drop=False, inplace=False):
                 out[name] = self.index._values
             for col_name, col_value in self._data.items():
                 out[col_name] = col_value
+            if isinstance(self.columns, cudf.core.multiindex.MultiIndex):
```

Comment:
We really need to make this a Pandas MultiIndex in the future... but in a future PR

Reply:
Are you wanting to drop `cudf.MultiIndex` entirely, or just when used as `columns`?

Reply:
Just when used as columns. We need to access these on the host as opposed to on the device so it doesn't make sense to use an on device MultiIndex.

Future PR though as this handling of multiindex for columns is kinda fragile as it is.